File: server/api/extras.ts
import { triggerPromise } from '@trippler/tr_lib/server'
import { VehicleClass } from '../../shared/types'

on('entityCreated', async (handle: number) => {
  const source = NetworkGetEntityOwner(handle)
  const vehicleClass = await triggerPromise<VehicleClass>('getVehicleClass', source)

  const entityType = GetEntityType(handle) != null ? GetEntityType(handle) : null
  if (entityType && entityType == 2 && vehicleClass && vehicleClass == 18) {
    setTimeout(() => {
      emitNet('lenix_patrolextras:client:checkVehicleExtras', source, handle, vehicleClass)
    }, 500)
  }
})

File: lenixfx/lenix_patrol/server/api/extras.lua
AddEventHandler('entityCreated', function(handle)
  local source<const> = NetworkGetEntityOwner(handle)
  
  CreateThread(function()
    local success, vehicleClass = lib.triggerPromise('getVehicleClass', source)
    local entityType<const> = GetEntityType(handle)

    if entityType == 2 and vehicleClass == 18 then
      SetTimeout(500, function()
        TriggerClientEvent('lenix_patrolextras:client:checkVehicleExtras', source, handle, vehicleClass)
      end)
    end
  end)
end)